core/infrastructure/imagestream.go

```diff
@@ -83,32 +102,103 @@ func (i *imageStreamHandler) MustFetchImageStream(key types.NamespacedName) (*im
 	}
 }
 
-// CreateImageStream creates the ImageStream referencing the given namespace.
+func (i *imageStreamHandler) CreateImageStreamIfNotExists(name, namespace, tag string, addFromReference bool, imageName string, insecureImageRegistry bool) (*imgv1.ImageStream, error) {
+	imageStream, err := i.FetchImageStream(types.NamespacedName{Name: name, Namespace: namespace})
+	if err != nil {
+		return nil, err
+	}
+	if imageStream == nil {
+		imageStream = i.createImageStream(name, namespace)
+	}
+
+	isTagExists := i.checkIfTagExists(imageStream, tag)
+	if !isTagExists {
+		tagReference := i.createImageStreamTag(tag, addFromReference, imageName, insecureImageRegistry)
+		tagReferences := append(imageStream.Spec.Tags, tagReference)
+		imageStream.Spec.Tags = tagReferences
+	}
+	return imageStream, nil
+}
+
+// createImageStream creates the ImageStream referencing the given namespace.
+func (i *imageStreamHandler) createImageStream(name, namespace string) *imgv1.ImageStream {
+	i.Log.Debug("Creating new Image stream.")
```

Comment:
please add the name of the created img stream

Reply:
done